internal/updater/updater.go

```diff
@@ -68,8 +68,11 @@ func (u *Updater) Apply(ctx context.Context, b *olmv1alpha1.Bundle) error {
 func EnsureCondition(condition metav1.Condition) UpdateStatusFunc {
 	return func(status *olmv1alpha1.BundleStatus) bool {
 		existing := meta.FindStatusCondition(status.Conditions, condition.Type)
-		meta.SetStatusCondition(&status.Conditions, condition)
-		return existing == nil || !conditionsSemanticallyEqual(*existing, condition)
```

Comment:
Note: this logic is only problematic when we have an existing condition that matches the condition.Type (e.g. `Unpacked`), so we prematurely set the status conditions before checking whether the existing and new conditions are semantically equal.